features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/h2.sql

```diff
@@ -1501,3 +1522,8 @@ CREATE INDEX IDX_IDN_ACTION_ENDPOINT_AU_TI ON IDN_ACTION_ENDPOINT (ACTION_UUID,
 -- CERTIFICATE --
 CREATE INDEX IDX_IDN_CERTIFICATE_ID_TID ON IDN_CERTIFICATE (ID, TENANT_ID);
 CREATE INDEX IDX_IDN_CERTIFICATE_UUID_TID ON IDN_CERTIFICATE (UUID, TENANT_ID);
+
+-- RULES --
+CREATE INDEX IDX_IDN_RULE_UUID_TID ON IDN_RULE (UUID, TENANT_ID);
+CREATE INDEX IDX_IDN_RULE_REF_RID_TID ON IDN_RULE_REFERENCES (RULE_ID, TENANT_ID);
+CREATE INDEX IDX_IDN_RULE_REF_FN_FREF_TID ON IDN_RULE_REFERENCES (FIELD_NAME, FIELD_REFERENCE, TENANT_ID);
```

Comment:
Why we need this index?

Reply:
This index is needed as we need to find rules based on fields and field values once resources associated for those fields and filed values can get deleted (e.g:, an application that is referred in a rule gets deleted).

However, at this point of time even though the index is present the query or the respective funtionality that would utilize this is not present.
So let's add it back when needed

Reply:
Removed the redundant index at this point of time based on the discussion above from commit: https://github.com/wso2/carbon-identity-framework/pull/6204/commits/c904b60008bc3efb85b9a7f46790b3c380649924